app/src/main/java/openfoodfacts/github/scrachx/openfood/features/shared/views/TipBox.kt

```diff
@@ -53,14 +68,32 @@ class TipBox(context: Context, attrs: AttributeSet?) : LinearLayout(context, att
             prefs.edit { putBoolean(identifier, false) }
         }
 
-        binding.tipBoxContainer.setBackgroundColor(toolTipBackgroundColor)
+        binding.closeBoxIcon.setOnClickListener {
+            hide()
+            prefs.edit { putBoolean(identifier, false) }
+        }
+
+        binding.mail.setOnClickListener {
+            val contactIntent = Intent(Intent.ACTION_SENDTO)
+            contactIntent.data = Uri.parse("mailto:contact@openfoodfacts.org")
+            contactIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+            try {
+                context.startActivity(contactIntent)
+            } catch (e: ActivityNotFoundException) {
+                Toast.makeText(getContext(), R.string.email_not_found, Toast.LENGTH_SHORT).show()
+            }
+        }
+
 
         if (canDisplayImmediately) loadToolTip()
     }
 
 
     fun setTipMessage(message: CharSequence?) {
-        binding.tipMessage.text = message
+        if (identifier != "showEcoScorePrompt")
```

Comment:
Put `binding.tipMessageButton.text` before the if

Reply:
This actually was done keeping in mind that the new TipBox Layout might be used somewhere else in the future. and if a setTipMessage is called, this can be edited to set the message to the desired TipBox layout. We can change ```fun setTipMessage``` to as it was before too as it is not being used to set message to the new layout yet.